contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -283,22 +292,39 @@ def _transcribe(item, parents, editRate, masterMobs):
         # The goal here is to find an available range. Media ranges are stored
         # in the related MasterMob, and there should only be one - hence the name
         # "Master" mob. Somewhere down our chain (either a child or our parents)
-        # is a MasterMob. For SourceClips in the CompositionMob, it is our child.
-        # For everything else, it is a previously encountered parent. Find the
-        # MasterMob in our chain, and then extract the information from that.
-        child_mastermob = (
-            item.mob if isinstance(item.mob, aaf2.mobs.MasterMob) else None
-        )
+        # is a MasterMob.
+        # There are some special cases where the masterMob could be:
+        # 1) For SourceClips in the CompositionMob, the mob the SourceClip is
+        #    referencing can be our MasterMob.
+        # 2) If the source clip is referencing another CompositionMob,
+        #    drill down to see if the composition holds the MasterMob
+        # 3) For everything else, it is a previously encountered parent. Find the
+        #    MasterMob in our chain, and then extract the information from that.
+
+        child_mastermob, compositionUserMetadata = _find_mastermob_for_sourceclip(item)
```

Comment:
I think its probably a good idea to make sure this adapter also adheres more closely to pep8:

```suggestion
        child_mastermob, composition_user_metadata = _find_mastermob_for_sourceclip(item)
```

Reply:
yes, I'll change that.